### PR TITLE
validations and model specs

### PR DESCRIPTION
**Pull request by HSMathebula**
In this pull request I :

- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validation

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  it 'should list three recent posts' do
+    4.times do |i|
+      Post.create(title: 'Hello', text: "This is my #{i}th post", comments_counter: 0, likes_counter: 0,
+                  author: subject)
+    end
+
+    expect(subject.three_recent_posts[0].text).to eq 'This is my 3th post'
+    expect(subject.three_recent_posts.length).to eq 3
+  end
+end
```
Comment (DeliceLydia) [inline spec/models/user_spec.rb:31]: - [ ] Please fix your tests that are failing
s

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @HSMathebula, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights are:
- Good README
- Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/models/user_spec.rb (hunk 1 above)
Comment (HSMathebula) [reply spec/models/user_spec.rb:31]: Hi @DeliceLydia, I ran the test and they all succeed 
here is a screenshot of my test.
image

**Review by phelian23 — APPROVED** [review]
Hi @HSMathebula, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f5061536e555a573145657970756c734f77642f67697068792e676966


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**